test/Feature/HLSLLib/reflect.16.test

```diff
@@ -0,0 +1,209 @@
+#--- source.hlsl
+
+// This test tests three different reflection scenarios:
+// -----> |
+// -----> /
+// -----> -
+// each test case will cycle through 1-4 dimensions.
+
+StructuredBuffer<half4> IncidentRay : register(t0);
+StructuredBuffer<half4> Wall1 : register(t1);
+StructuredBuffer<half4> Wall2 : register(t2);
+StructuredBuffer<half4> Wall3 : register(t3);

```

Comment:
You're using StructuredBuffers, which means you can define everything you need in one buffer for simplicity, right?  Same with the output buffer.